compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -1258,9 +1262,26 @@ object Trees {
         transform(tree).asInstanceOf[Tr]
       def transformSub[Tr <: Tree](trees: List[Tr])(implicit ctx: Context): List[Tr] =
         transform(trees).asInstanceOf[List[Tr]]
+
+      protected def transformMoreCases(tree: Tree)(implicit ctx: Context): Tree = tree match {
+        case tpd.UntypedSplice(usplice) =>
+          // For a typed tree map: homomorphism on the untyped part with
+          // recursive mapping of typed splices.
+          // The case is overridden in UntypedTreeMap.##
```

Comment:
There is no `TypedTreeMap` and this addition did not seem sufficient reason to me to create one.  Mostly because I was worried that creating more indirections in tree maps would hurt performance.